src/common/TooltipInfo/TooltipInfo.tsx

```diff
@@ -1,14 +1,50 @@
 import * as React from "react";
+import { checkExhaustively } from "../../utilities";
+import { toast } from "bulma-toast";
+
+export enum TooltipDirection {
+    TOP = "TOP",
+    BOTTOM = "BOTTOM",
+    LEFT = "LEFT",
+    RIGHT = "RIGHT"
+}
 
 export interface TooltipInfoProps {
     text: string;
     url?: string;
+    direction?: TooltipDirection;
+}
+
+function getClassName(direction: TooltipDirection | undefined): string {
+    const base = "icon has-tooltip-multiline has-tooltip-arrow";
+
+    switch (direction) {
+        case undefined:
```

Comment:
This is a strange "fall through". Isn't this handled by "default"?

Reply:
It could be handled by default, but I chose to rather handle it specifically and use checkExhaustively in case we modify the enum in the future and forget updating the switch.

Reply:
Can you address why the undefined part is here at all? I really don't like this behaviour in code. Though to be fair I am increasingly against switches and if-else statements, too, due to their impact on the open closed principle.

Reply:
Undefined is there because direction is an optional property. When the user does not specify any direction, the default direction should be TOP, thus both undefined and TOP have the same code.

Undefined cannot be caught in the default case, as undefined does not match the never in checkExhaustively.

Reply:
if (!direction) return (default enum here) prior to the switch would handle that much better and more explicitly.

Reply:
Could alternatively do something like `const finalDirection = direction || TooltipDirection.TOP`.